contexts-state-store/api/golang/kurtosis_context_visitor.go

```diff
@@ -0,0 +1,22 @@
+package golang
+
+import (
+	"github.com/kurtosis-tech/kurtosis/contexts-state-store/api/golang/generated"
+	"github.com/kurtosis-tech/stacktrace"
+	"reflect"
+)
+
+type KurtosisContextVisitor[ResultType any] struct {
+	VisitLocalOnlyContextV0 func(localContext *generated.LocalOnlyContextV0) (*ResultType, error)
+
+	VisitRemoteContextV0 func(localContext *generated.RemoteContextV0) (*ResultType, error)
+}
+
+func Visit[ResultType any](kurtosisContext *generated.KurtosisContext, visitor KurtosisContextVisitor[ResultType]) (*ResultType, error) {
+	if kurtosisContext.GetLocalOnlyContextV0() != nil {
+		return visitor.VisitLocalOnlyContextV0(kurtosisContext.GetLocalOnlyContextV0())
```

Comment:
why do we have the V0 suffix?

Reply:
To have a V1 in the near future?? :P I imagine those objects are going to evolve, and we can do so very easily by adding a new member to the union type. If we want to make a breaking change to `LocalOnlyContextV0`, we can introduce `LocalOnlyContextV1` by adding it to the oneof proto file

Reply:
feels a bit like future proofing but we already have it so keep it :) Resolving this.